### PR TITLE
migrate sprintf to null-safety

### LINKED ISSUE #24: Migrate sprintf to null-safety
Looks like this package is quite popular and used by many (including me). Decided to make a PR migrating it to Dart's new language feature

https://medium.com/dartlang/announcing-dart-null-safety-beta-87610fee6730

### PR DESCRIPTION
**Pull request by vishna**
closes #24

### CODE REVIEW

**Comment by vishna** [comment]
CI fails on stable which is __expected__ because of the dart version.

File: lib/src/formatters/Formatter.dart [hunk 1]
```diff
@@ -21,5 +21,5 @@ abstract class Formatter {
     return padding.toString();
   }
 
-  String asString();
+  String? asString();
```
Comment (Naddiseo) [inline lib/src/formatters/Formatter.dart:24]: I don't think this change is needed

File: lib/src/formatters/float_formatter.dart [hunk 2]
```diff
@@ -93,7 +93,7 @@ class FloatFormatter extends Formatter {
   }
 
   @override
-  String asString() {
+  String? asString() {
```
Comment (Naddiseo) [inline lib/src/formatters/float_formatter.dart:96]: This change isn't needed. The `_output` var has a direct null check before it's returned, so the line can be changed to ` return _output!;`

File: lib/src/sprintf_impl.dart [hunk 3]
```diff
@@ -85,7 +85,7 @@ class PrintFormat {
         }
         _arg_str = '%';
       } else if (_formatters.containsKey(_type)) {
-        _arg_str = _formatters[_type](_arg, _options).asString();
+        _arg_str = _formatters[_type]!(_arg, _options).asString()!;
```
Comment (Naddiseo) [inline lib/src/sprintf_impl.dart:88]:
```suggestion
        _arg_str = _formatters[_type]!(_arg, _options).asString();
```

**Review by Naddiseo — CHANGES_REQUESTED** [review]
Thanks for the PR, mostly looks good, two other files need changing:
1. Could you update .travil.yml to replace "stable" with "beta"
2. Could you add an entry into the CHANGELOG file

thanks!

**Comment by vishna** [comment]
updated my PR

**Pull request merged**